src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -51,6 +51,11 @@ internal class CompletionHandler : IRequestHandler<LSP.CompletionParams, LSP.Com
                 return null;
             }
 
+            if (request.Context?.TriggerCharacter == null)
+            {
+                return null;
```

Comment:
TriggerCharacter is `null` in a proper functioning LSP client when typing out an identifier. I don't think returning `null` here is correct.
![image](https://user-images.githubusercontent.com/2008729/95044345-0e504480-0694-11eb-8d82-e23bac2cfe71.png)

Reply:
Also, to be clear, Razor translates the LSP platform's requests into spec abiding LSP requests which ends up hitting this call path. If you checked this in it'd stop returning C# identifier completions in Razor scenarios 😢

Reply:
I think for now we can just match the spec and say invoked if we get null.  If we actually get null, this will result in completions coming up unexpectedly (as invoked is generally more aggressive).  But VS is currently passing the trigger char 'incorrectly' so we'll only hit this in Razor right now.

Then we can followup with a longterm fix from https://github.com/microsoft/ms-lsp/issues/57#issuecomment-703863301 (probably some kind of new property)

Reply:
I've updated the PR to what I believe is spec-compliant behaviour, specifically in [`0f09454` (#47882)](https://github.com/dotnet/roslyn/pull/47882/commits/0f09454e5725a9737dcca46eb180d1279e13070b). PTAL.